src/blackmetal/src/communication/controlSW/Odometry.hpp

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <thread>
+class Odometry;
+#include "controlSW/BlackMetal.hpp"
+
+#include <rclcpp/timer.hpp>
+
+/**
+ * @class Odometry
+ * @brief Manages left and right wheel speed using the control server.
+ *
+ * With a set frequency sends requests to server and evaluates the position, of where it is located
+ * in the Cartesian plain. The beginning position after turning on 
+ * the robot is [0, 0, 0]. The coordinates are represented as [x, y, angle].
+ *
+ */
+class Odometry
+{
+public:
+	/**
+	 * @class Coord
+	 * @brief Specifies the coordinates and angle of rotation of the robot.
+	 */
+	struct Coord {
+		/// The x coordinate of the robot in Cartesian plain.
+		double x;
+		/// The y coordinate of the robot in Cartesian plain.
+		double y;
+		/// The angle of the robot with the x axis in Cartesian plain.
+		double angle;
+	};
+
+	/**
+	 * @class Speed
+	 * @brief Stores the speeds of the left and right wheel obtained from the robot.
+	 */
+	struct Speed
+	{
+		/// The velocity of left wheel.
+		long leftWheel;
+		/// The velocity of right wheel.
+		long rightWheel;
+	};
+
+	/**
+	 * @brief Creates an odometry object. This object polls the server for its velocity.
+	 * Therefor the object should call the execute function in different thread.
+	 *
+	 * @param controlSoftware Reference to the control software connected to the robot.
+	 */
+	explicit Odometry(BlackMetal &controlSoftware);
+
+	/**
+	 * @brief Function polling the robot for its left and right wheel speed.
+	 * The function will block the thread, thus it should be run in different thread.
+	 */
+	void execute();
+
+	/**
+	 * @brief Retrieve the left and right wheel speed from the received json message.
+	 *
+	 * @param jsonMessage Message received from the server.
+	 * @return Structure of left and right wheel speed.
+	 */
+	Speed obtainWheelSpeeds(const std::string &jsonMessage);
```

Comment:
povedal by som, zeby asi mohla byt const
```suggestion
	Speed obtainWheelSpeeds(const std::string &jsonMessage) const;
```